subsys/bluetooth/host/att.c

```diff
@@ -276,14 +278,29 @@ static int chan_send(struct bt_att_chan *chan, struct net_buf *buf)
 	return err;
 }
 
+static bool att_chan_matches_chan_pref(struct bt_att_chan *chan, enum bt_att_chan_pref chan_pref)
+{
+	__ASSERT_NO_MSG(chan_pref < 0x03);
+
+	if (chan_pref == BT_ATT_CHAN_ANY) {
+		return true;
+	}
+
+	if (atomic_test_bit(chan->flags, ATT_ENHANCED)) {
+		return (chan_pref == BT_ATT_CHAN_ENHANCED);
+	} else {
+		return (chan_pref == BT_ATT_CHAN_UNENHANCED);
+	}
```

Comment:
The `chan_pref` (or option) value is a bitfield. So if `chan_pref = BT_ATT_CHAN_ENHANCED | BT_ATT_CHAN_UNENHANCED`, this check fails. Is that intentional?

Reply:
It was intentional (see assert above), as "any" is now `0x0`. 

Do we want to allow both `0x0` and `0x3` to be "any"? Then the bit-field would make sense at the cost of two options having the same meaning.

Reply:
It's a good question. 

The bitfield approach will make it easier to add more fields to the enum without changing existing behavior (in theory at least), but it is kinda funky with the current options. 

Do we have a use case for wanting to send on an unenhanced ATT bearer? If not, then we could consider have `0x00` = "No preference"/"No options", and then only have a single field (for now):
`BT_ATT_CHAN_ENHANCED_ONLY = BIT(0)`

so that we only have an option to say "Enhanced only" which is what we have  use case for. So 0x00 will be any ATT bearer, and 0x01 (`BIT(0)`) will be EATT only.

Reply:
There are some PTS tests that require us to send on unenhanced, though I am not sure if any "real" application needs it. Having a flag only for "EATT only" aligns with the audio use-case, but is less general.

Reply:
> There are some PTS tests that require us to send on unenhanced

That's interesting. So in those tests we are allowed to connect EATT channels, but not allowed to use them? That seems to require either a test spec update or a PTS erratum, rather than an API choice for Zephyr.

Reply:
See eg. GATT/CL/GPM/BV-11-C and friends, which tests client side GATT caching. There the test procedure requires the IUT to first send a request on an unenhanced bearer, then on an enhanced bearer.